src/Context/AuthContext.js

```diff
@@ -0,0 +1,82 @@
+import axios from "axios";
+import { createContext, useContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext({
+  signup: () => {},
+  login: () => {},
+ token:"",
+  user:"",
+  logout:()=>{},
+  isLoggedIn:""
+});
+
+const AuthProvider = ({ children }) => {
+    const navigate = useNavigate();
+   const tokenInLocalStorage= localStorage.getItem("token")
+   const userInLocalStorage = localStorage.getItem("user")
```

Comment:
Can use ternary operators here just to make sure the variables have some initial fallback value to go to instead of undefined or null if there is no such variable in localStorage.